### PR TITLE
docs: Improves explanation of caching and cache volumes

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit adds more information about caching to the Dagger Cloud and Dagger quickstarts and the FAQ

### CODE REVIEW

File: docs/current/faq.md [hunk 1]
```diff
@@ -151,6 +151,15 @@ Follow these steps:
     - Python: [https://github.com/platformdirs/platformdirs](https://github.com/platformdirs/platformdirs)
   :::
 
+### What are Dagger's caching features?
```
Comment (vikram-dagger) [inline docs/current/faq.md:154]: This is a first cut, would like inputs from @gerhard @vito @jpadams @marcosnils @shykes @samalba on this one.

**Review by TomChv — APPROVED** [review]

**Review by shykes — APPROVED** [review]
Looks great! Thanks for doing this.

File: docs/current/faq.md (hunk 1 above)
Comment (gerhard) [reply docs/current/faq.md:154]: Looks good to me @vikram-dagger 👍

Will merge & publish with a bunch of other docs changes that are still in flight.

**Review by gerhard — APPROVED** [review]
👍 🚀

**Pull request merged**